cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapperWrapper.java

```diff
@@ -195,15 +204,18 @@ public InputSplit getInputSplit() {
     return flushingContext;
   }
 
-  private Mapper createMapperInstance(ClassLoader classLoader, String userMapper, Context context) {
+  private Mapper createMapperInstance(ClassLoader classLoader, String userMapper, Context context, String program) {
     if (context.getInputSplit() instanceof MultiInputTaggedSplit) {
       // Find the delegate Mapper from the MultiInputTaggedSplit.
       userMapper = ((MultiInputTaggedSplit) context.getInputSplit()).getMapperClassName();
     }
     try {
       return (Mapper) classLoader.loadClass(userMapper).newInstance();
     } catch (Exception e) {
+      Throwable rootCause = Throwables.getRootCause(e);
```

Comment:
do you have an example of such an exception? I am not convinced that we need the root cause